aiohttp/_websocket/reader.py

```diff
@@ -1,350 +1,21 @@
 """Reader for WebSocket protocol versions 13 and 8."""
 
-from enum import IntEnum
-from typing import Final, List, Optional, Set, Tuple
+from typing import TYPE_CHECKING
 
-from ..compression_utils import ZLibDecompressor
-from ..helpers import set_exception
-from ..streams import DataQueue
-from .helpers import UNPACK_CLOSE_CODE, UNPACK_LEN2, UNPACK_LEN3, websocket_mask
-from .models import (
-    WS_DEFLATE_TRAILING,
-    WebSocketError,
-    WSCloseCode,
-    WSMessage,
-    WSMessageBinary,
-    WSMessageClose,
-    WSMessagePing,
-    WSMessagePong,
-    WSMessageText,
-    WSMsgType,
-)
+from ..helpers import NO_EXTENSIONS
 
-MESSAGE_TYPES_WITH_CONTENT: Final = frozenset(
-    {
-        WSMsgType.BINARY,
-        WSMsgType.TEXT,
-        WSMsgType.CONTINUATION,
-    }
-)
+if TYPE_CHECKING or NO_EXTENSIONS:  # pragma: no cover
+    from .reader_py import WebSocketReader as WebSocketReaderPython
 
-ALLOWED_CLOSE_CODES: Final[Set[int]] = {int(i) for i in WSCloseCode}
+    WebSocketReader = WebSocketReaderPython
+else:
+    try:
+        from .reader_c import (  # type: ignore[import-not-found]
+            WebSocketReader as WebSocketReaderCython,
+        )
 
+        WebSocketReader = WebSocketReaderCython
+    except ImportError:  # pragma: no cover
+        from .reader_py import WebSocketReader as WebSocketReaderPython
 
-class WSParserState(IntEnum):
-    READ_HEADER = 1
-    READ_PAYLOAD_LENGTH = 2
-    READ_PAYLOAD_MASK = 3
-    READ_PAYLOAD = 4
-
-
-class WebSocketReader:
-    def __init__(
-        self, queue: DataQueue[WSMessage], max_msg_size: int, compress: bool = True
-    ) -> None:
-        self.queue = queue
-        self._max_msg_size = max_msg_size
-
-        self._exc: Optional[BaseException] = None
-        self._partial = bytearray()
-        self._state = WSParserState.READ_HEADER
-
-        self._opcode: Optional[int] = None
-        self._frame_fin = False
-        self._frame_opcode: Optional[int] = None
-        self._frame_payload = bytearray()
-
-        self._tail: bytes = b""
-        self._has_mask = False
-        self._frame_mask: Optional[bytes] = None
-        self._payload_length = 0
-        self._payload_length_flag = 0
-        self._compressed: Optional[bool] = None
-        self._decompressobj: Optional[ZLibDecompressor] = None
-        self._compress = compress
-
-    def feed_eof(self) -> None:
-        self.queue.feed_eof()
-
-    def feed_data(self, data: bytes) -> Tuple[bool, bytes]:
```

Comment:
This has been typed wrong for a very long time since Protector uses bytearray and not bytes.

Technically it could be a memoryview as well but that seems unlikely